### PR TITLE
Add basic functionality of handle:

### LINKED ISSUE #18: Create implementation for handle function
Please, create implementation for handle function. This function should:

1. Get all suitable sections from parseSection function
2. Send it to splitSection function, which will split sections into options
3. Get a two-dimensional array of all options from parseOption function

### PR DESCRIPTION
**Pull request by ifedchankau**
- Add implementation of handle function
- Add functions parseSection and parseOption (w/o implementation)

Closes #18

### CODE REVIEW

File: src/handle.js [hunk 1]
```diff
@@ -1,6 +1,25 @@
 const handle = (help) => {
-    // TODO: create implementation of parse function
+    const parseSection = require('./parseSection.js');
+    const parseOption = require('./parseOption');
+    const sectionName = "Options:";
+
     let options = [];
+    parseSection(sectionName, help).forEach(section => {
+        let splitTmp = (section.split(/\n[ \t]*(-\S+?)/)).slice(1);
```
Comment (xferra) [inline src/handle.js]: Unnecessary `()` and move regex value to a variable, ideally that has a descriptive name what it does.

File: src/handle.js [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+        let split = [];
+        for (let cnt = splitTmp.length, i = 0; i < cnt; i+=2) {
```
Comment (xferra) [inline src/handle.js]: I suggest to isolate it in a function like `splitPairs`.

File: src/handle.js [hunk 3]
```diff
@@ -1,6 +1,25 @@
...
+            if (splitTmp !== undefined) {
+                split.push(splitTmp[i] + splitTmp[i + 1]);
```
Comment (xferra) [inline src/handle.js]: Probably `[ splitTmp[i] , splitTmp[i + 1] ]`?

File: src/handle.js [hunk 4]
```diff
@@ -1,6 +1,25 @@
...
+            } else {
+                split.push(splitTmp[i]);
+            }
+        }
+        split.forEach(option => {
```
Comment (xferra) [inline src/handle.js]: Just FYI, alternative variant:
```javascript
split.filter((x) => x. startsWith('i')).map((x) => parseOption(section))
```

File: src/parseOption.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+const parseOption = (optionDescription) => {
```
Comment (xferra) [inline src/parseOption.js]: I hope there is no need to create separate modules and files for each function. You can export an object as well:
```javascript
const parseOption = () => {};
const parseSection = () => {};

module.exports = {
   parseOption: parseOption,
   parseSection: parseSection
};

```

Comment (xferra) [reply src/parseOption.js]: +Consider using [classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes), JS supports not the only [procedural paradigm](https://en.wikipedia.org/wiki/Procedural_programming).

File: src/handle.js (hunk 3 above)
Comment (xferra) [reply src/handle.js]: I'm trying to follow the logic here, split then concat back, what is the idea?

Comment (ifedchankau) [reply src/handle.js]: Sorry, i forgot about set flag PREG_SPLIT_DELIM_CAPTURE of preg_split php function when transating it to js. If this flag is set, parenthesized expression in the delimiter pattern will be captured and returned as well. This is also an answer on your question about parenthes in regular expression.

For example, with reg exp /\n[ \t]*(-\S+?)/ string
"-c, --config          configuration file                                                                                
  -e, --exclude         exclude globs from path expansion"                                                              
will be sliced into 4 strings:  
* "-c" 
* ", --config          configuration file"
* "-e" 
* ", --exclude         exclude globs from path expansion"                                                           

So, its needed to concate 1 and 2, 3 and 4 strings later

Maybe you know, what is js alternative for PREG_SPLIT_DELIM_CAPTURE flag or can suggest another solution of this?

Comment (xferra) [reply src/handle.js]: Like this one?
https://searchcode.com/codesearch/view/14390265/#

There is no alternative out of the box.

**Review by xferra — APPROVED** [review]
+1

**Pull request merged**